include/boost/beast/core/impl/buffered_read_stream.ipp

```diff
@@ -142,11 +144,11 @@ buffered_read_stream(Args&&... args)
 
 template<class Stream, class DynamicBuffer>
 template<class ConstBufferSequence, class WriteHandler>
-auto
+BOOST_ASIO_INITFN_RESULT_TYPE(WriteHandler, void(error_code))
```

Comment:
The signature of an async_write_some handler of an AysncWriteStream should be void(error_code,size_t).